chucks-pick3-web/angular-app/src/app/components/customer-summary.component/customer-summary.component.ts

```diff
@@ -46,19 +48,78 @@ export class CustomerSummaryComponent implements OnInit {
   public addRow() {
     const customer = this.customers;
     // TODO: create a new customer.
-    const johnBowman = {
-      firstName: 'John',
-      lastName: 'Bowman',
-      email: 'jbowman@gmail.com',
+    const newCust = {
+      firstName: this.generateFirstName(),
+      lastName: this.generateLastName(),
+      email: this.generateEmail(),
       phone: '555-555-5555',
-      city: 'Chicago',
-      state: 'Illinois',
-      id: 76
+      city: this.generateCity(),
+      state: this.generateState(),
+      id: this.generateId()
     };
     // TODO: const aCustomer = this.generateCustomer()
-
     // TODO: add new customer to array.
-    customer.push(johnBowman);
+    customer.push(newCust);
     /*   });*/
   }
+
+  private generateFirstName() {
+    const firstName =
+      [
+        'Harvey',
+        'Murray',
+        'Jack'
+      ];
+    return firstName[Math.floor(Math.random() * 4)];
+  }
+
+  private generateLastName() {
+    const lastName =
+      [
+        'Pliskin',
+        'Smith',
+        'Gallegos'
+      ];
+    return lastName[Math.floor(Math.random() * 4)];
```

Comment:
Don't hard code the max value 4.  Use the array.length; so that it will be dynamic.

Reply:
Made max value dynamic.